File: app/src/main/java/com/training/android/materiel/ui/activity/RegisterContactActivity.java
package com.training.android.materiel.ui.activity;

import android.app.ActionBar;
import android.app.Activity;
import android.support.v4.app.NavUtils;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import com.training.android.materiel.R;

public class RegisterContactActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register_contact);
        ActionBar toolbar = getActionBar();
        if (toolbar != null) {
            toolbar.setDisplayShowTitleEnabled(false);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_validate, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == android.R.id.home) {
            NavUtils.navigateUpFromSameTask(this);
        } else if (id == R.id.action_validate) {
            // TODO: Perform action.
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/components/text_fields/RegisterContactActivity.kt
package fr.erictruong.material.training.ui.components.text_fields

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import fr.erictruong.material.training.R
import fr.erictruong.material.training.extension.consume
import org.jetbrains.anko.toast

class RegisterContactActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register_contact)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.validate, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_validate -> consume { onActionValidate() }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun onActionValidate() {
        // TODO
        toast("Placeholder action")
        finish()
    }
}
